# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLFormSubmission.cpp
# Commit: 7e73e0ab27fe
# Full Hash: 7e73e0ab27feb43b078eabb317ed191c878efc9f
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2020-07-15 09:37:18
# Description:
#   Bug 1652699 - Fix a null pointer crash while doing dialog form submission r=smaug
#   
#   It's possible that the dialog is closed while we are processing
#   the request, and we'd deferencing a null pointer for such case.
#   
# ==============================================================================

diff -r aaa0c6fdc568 -r 7e73e0ab27fe dom/html/HTMLFormSubmission.cpp
--- a/dom/html/HTMLFormSubmission.cpp	Tue Jul 14 12:20:10 2020 +0000
+++ b/dom/html/HTMLFormSubmission.cpp	Tue Jul 14 21:19:10 2020 +0000
@@ -830,7 +830,7 @@
     // If there isn't one, or if it does not have an open attribute, do
     // nothing.
     if (!dialog || !dialog->Open()) {
-      return NS_OK;
+      return NS_ERROR_FAILURE;
     }
 
     nsAutoString result;